crates/core/c8y_api/src/smartrest/operations.rs

```diff
@@ -165,6 +177,35 @@ impl Operations {
         None
     }
 
+    pub fn filter_for_json_input(&self, topic_name: &str) -> Vec<Operation> {
+        let mut vec: Vec<Operation> = Vec::new();
+        for op in self.operations.clone() {
+            if op.exec.is_none() {
+                continue;
+            }
+
+            match op.topic() {
+                Some(topic) if topic != topic_name => {}
+                _ => {
+                    if op.on_message().is_some() {
+                        warn!(
+                            "'on_message' should not be provided in the custom operation handler mapping '{}', which is for SmartREST input.",
+                            op.name
+                        );
+                    } else if op.on_fragment().is_none() {
+                        warn!(
+                            "'on_fragment' is missing in the custom operation handler mapping '{}'",
+                            op.name
+                        );
+                    } else {
```

Comment:
I was actually also considering the way. I'll try to implement a validation function quickly after I finish doc. It should cover other patterns as well.

For all:
- files without `[exec]` field
For SmartREST handlers:
- `topic` is missing
- `on_message` is missing
- `on_fragment` is given
- `command` is missing
For JSON handlers:
- `on_message` is given
- `on_fragment` is missing
- `command` is missing (this will be changed once "drop 2" is completed

The good thing of a centralized validation function is, when we need to add something new, developer just has to revisit the function. If validation is made in distributed functions, one needs to revisit all of them.

Reply:
Done in f9ef26492cd7f7b264ce652b4998c626f4b1926c